rustls/examples/internal/bogo_shim.rs

```diff
@@ -200,7 +200,7 @@ impl server::danger::ClientCertVerifier for DummyClientAuth {
         self.mandatory
     }
 
-    fn client_auth_root_subjects(&self) -> &[DistinguishedName] {
+    fn client_auth_hint_root_subjects(&self) -> &[DistinguishedName] {
```

Comment:
How about we get rid of the client_auth_ prefix for these methods? The context seems fairly obvious from the trait name.

I also don't love the rest of this name. Maybe we can skip subjects as well, given that the type is DistinguishedName? root_hints() seems like a pretty good name.

Reply:
> How about we get rid of the client_auth_ prefix for these methods?

Done :heavy_check_mark: 

> I also don't love the rest of this name. Maybe we can skip subjects as well, given that the type is DistinguishedName? root_hints() seems like a pretty good name.

I went with `root_hint_subjects` and standardized on that terminology throughout. I think keeping the "subjects" part is useful since `DistinguishedName`'s appear in other contexts as well (issuer, for example).